Sources/Interpreter/Memory.swift

```diff
@@ -357,3 +372,99 @@ public extension Memory.Address {
   static func -=(l: inout Self, r: Int)  { l = l - r }
 
 }
+
+extension Memory.Allocation {
+
+  /// Returns initaialized region containing offset `o`.
+  func composedRegion(containingOffset o: Int) -> ComposedRegion {
+    precondition(o + baseOffset < storage.count)
+    let i = composedRegions.partitioningIndex { $0.offset > o } - 1
+    return composedRegions[i]
+  }
+
+  /// Returns the immediate subobject of `t` which contains offset `o` in `self`,
```

Comment:
```suggestion
  /// Returns the immediate subobject of `t` that contains offset `o` in `self`,
```

“which” is only appropriate after a comma.  Subtle rule of English; it's because of meaning, not punctuation, but the comma is a good guide as to which meaning you want.